metabase/metabase
# Allow archiving query actions

Epic #27581 

Makes it possible to archive and unarchive query actions (basic/implicit actions are out of scope for now).

### How to verify

1. Go to `/model/:id/details/actions`
2. Create a query action if the model doesn't have any yet
3. Click the ellipsis icon next to the action name, click the "Archive" option, and confirm
4. Ensure you can see a toast and can undo the operation
5. Go to `/archive`
6. Ensure you can unarchive the action and see it on the model detail page again
7. Ensure you can permanently delete the action from the archive page

### Demo

https://user-images.githubusercontent.com/17258145/219464048-68af50e9-4e76-4110-ba92-22da2206fa71.mp4

### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/28384)
